src/custom-lists/components/overview/sidebar/Index.js

```diff
@@ -0,0 +1,241 @@
+        if (this.inputBlockPattern.test(value)) {
+            return
+        }
+
+        this.setState(state => ({
+            ...state,
+            [field]: value,
+        }))
+    }
+
+    getSearchVal = value => value.trim().replace(/\s\s+/g, ' ')
+
+    handleCreateListSubmit = event => {
+        event.preventDefault()
+        const { value } = event.target.elements['listName']
+        // value = list name
+        this.props.createPageList(
+            this.getSearchVal(value),
+            this.vacateInputField,
+        )
+    }
+
+    vacateInputField = () => {
+        this.setState({
+            listName: null,
+        })
+    }
+
+    handleUpdateList = ({ id }, index) => event => {
+        event.preventDefault()
+        const { value } = event.target.elements['listName']
+        // value = list name
+        this.props.updateList(index, this.getSearchVal(value), id)
+        this.setState({
+            updatedListName: null,
+        })
+    }
+
+    renderAllLists = () => {
+        return this.props.lists.map((list, i) => {
+            if (list.isEditing) {
+                return (
+                    <CreateListForm
+                        key={i}
+                        onCheckboxClick={this.handleUpdateList(list, i)}
+                        handleNameChange={this.handleSearchChange(
+                            'updatedListName',
+                        )}
+                        value={
+                            typeof this.state.updatedListName === 'string'
+                                ? this.state.updatedListName
+                                : list.name
+                        }
+                        showWarning={this.state.showWarning}
+                        setInputRef={this.setInputRef}
+                    />
+                )
+            }
+            return (
+                <ListItem
+                    key={i}
+                    listName={list.name}
+                    isFiltered={list.isFilterIndex}
+                    onEditButtonClick={this.props.handleEditBtnClick(i)}
+                    onListItemClick={this.props.handleListItemClick(list, i)}
+                    onAddPageToList={this.props.handleAddPageList(list, i)}
+                    onCrossButtonClick={this.props.handleCrossBtnClick(list, i)}
+                />
+            )
+        })
+    }
+
+    renderCreateList = (shouldDisplayForm, value = null) =>
+        shouldDisplayForm ? (
+            <CreateListForm
+                onCheckboxClick={this.handleCreateListSubmit}
+                handleNameChange={this.handleSearchChange('listName')}
+                value={this.state.listName}
+                showWarning={this.props.showCommonNameWarning}
+                setInputRef={this.setInputRef}
+                closeCreateListForm={this.props.closeCreateListForm}
+            />
+        ) : null
+
+    render() {
+        return (
+            <div>
```

Comment:
It would be nice to move all these standard HTML elements to proper view components. Ideally JSX in the container should just render view components.

I would also avoid these nested classless `<div>` wrappers, as they have no semantic value.